### PR TITLE
Connecting with an API

### PR DESCRIPTION
**Pull request by robinbenitezmora**
As a part of Microverse practice, this time we are sharing with our app the data by means an API in order to go over the concepts inititalized in former challenges.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,64 +1,74 @@
...
+    },
+  );
+  const data = await response.json();
+  const books = [];
+  Object.keys(data).map((key) => books.push({ item_id: key, ...data[key][0] }));
+  dispatch({
+    type: GET_BOOKS,
+    payload: books,
+  });
+};
```
Comment (victorokoroji) [inline src/redux/books/books.js:40]: - [ ] According to the project requirement, kindly use [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) for making API calls 👍🏽 .

File: src/redux/books/books.js [hunk 2]
```diff
@@ -1,64 +1,74 @@
...
+const removeBook = (book) => async (dispatch) => {
+  const id = book.item_id;
+  await fetch(`${API}/${id}`, {
+    method: 'DELETE',
+    headers: { 'Content-Type': 'application/json' },
+  });
+  // dispatch({
+  //   type: REMOVE_BOOK,
+  //   payload: id,
+  // });
```
Comment (victorokoroji) [inline src/redux/books/books.js]: - [ ] Kindly remove all unused commented code from your codebase to make your code clean and readable 👍🏽

**Review by victorokoroji — CHANGES_REQUESTED** [review]
Hi @robinbenitezmora, 

You have done a great job so far! 💪 👏
You still need to work on a few issues before moving on to the next project, but you're almost there!

## Top highlight 🎯

- The Pull Request is following the GitHub Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- The commit messages are meaningful ✅ 
- Linters are working properly ✅ 


## Required Changes ♻️

- [ ] Kindly remove all unused commented code from your codebase to make your code clean and readable 👍🏽 

- [ ] According to the project requirement, kindly use [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) for making API calls 👍🏽 

- [ ] .

_Check the comments under the review._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 1 above)
Comment (robinbenitezmora) [reply src/redux/books/books.js:40]: Hello @vickymarz , could you give a 'thunk' example in order to apply it. Thanks a lot

Comment (victorokoroji) [reply src/redux/books/books.js:40]: I think you should check out for createAsynctThunk in your microverse resource / learning materials or you check it online on redux toolkit dicumentation

File: src/components/library/Book-list.js [hunk 3]
```diff
@@ -1,16 +1,21 @@
...
 
 const BookList = () => {
   const books = useSelector((state) => state.books, shallowEqual);
-  const booksList = books.map((book) => (
-    <Book key={book.id} book={book} />
-  ));
+
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(getBooks());
```
Comment (Davidosky007) [inline src/components/library/Book-list.js:11]: * Good job using createAsyncThunk for your API calls but you have a new bug 🐛 now as when i create a book it doesn't show up on the page, kindly fix this error 🙏 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @robinbenitezmora 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
* createAsyncThunk used for creating thunk action creators 👍 
* Connect an existing API via Redux 👍 
* Good looking readme file 👍 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidosky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mahbubul14 — APPROVED** [review]
# Project Approved :trophy: :tada: 

Hi @robinbenitezmora , 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linter error.
:heavy_check_mark: Descriptive readme file.
:heavy_check_mark: Well implemented suggested changes.

### Optional suggestions :arrow_down_small: 

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

- [ ] The commit message should be written in the imperative tense and the first letter should also be capitalized. `Ex: "Fix bug" instead of "Fixed bug".` [For more details...](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**